File: Booking.java
public class Booking {

    // Private variables (Encapsulation)
    private int bookingId;
    private String bookingDate;
    private int seatNumber;
    private String bookingStatus;

    public Booking(int bookingId, String bookingDate,
                int seatNumber, String bookingStatus) {

        this.bookingId = bookingId;
        this.bookingDate = bookingDate;
        this.seatNumber = seatNumber;
        this.bookingStatus = bookingStatus; 
    }

    public int getBookingId() {
        return bookingId;
    }

    public String getBookingDate() {
        return bookingDate;
    }

    public int getSeatNumber() {
        return seatNumber;
    }

    public String getBookingStatus() {
        return bookingStatus;
    }

    public void setBookingStatus(String bookingStatus) {
        this.bookingStatus = bookingStatus;
    }

    public void displayBookingDetails() {

        System.out.println("\n----------- Booking Details -----------");
        System.out.println("Booking ID : " + getBookingId());
        System.out.println("Booking Date : " + getBookingDate());
        System.out.println("Seat Number : " + getSeatNumber());
        System.out.println("Booking Status : " + getBookingStatus());
    }
}

File: python/Booking.py
# Encapsulation

class Booking:

    # Constructor
    def __init__(self, bookingId, bookingDate, bookedBy, seatNumber, bookingStatus):
        self.__bookingId = bookingId
        self.__bookingDate = bookingDate
        self.__bookedBy = bookedBy
        self.__seatNumber = seatNumber
        self.__bookingStatus = bookingStatus

    # Getter methods
    def getBookingId(self):
        return self.__bookingId

    def getBookingDate(self):
        return self.__bookingDate

    def getbookedBy(self):
        return self.__bookedBy

    def getSeatNumber(self):
        return self.__seatNumber

    def getBookingStatus(self):
        return self.__bookingStatus

    # Setter method
    def setBookingStatus(self, bookingStatus):
        self.__bookingStatus = bookingStatus

    # Displays booking details
    def displayBookingDetails(self):
        print("\n----------- Booking Details -----------")
        print("Booking ID :", self.getBookingId())
        print("Booking Date :", self.getBookingDate())
        print("Booked By :", self.getbookedBy())
        print("Seat Number :", self.getSeatNumber())
        print("Booking Status :", self.getBookingStatus())